## Shopify Data Challenge
#### Question 1: Given some sample data, write a program to answer the following: click here to access the required data set
* On Shopify, we have exactly 100 sneaker shops, and each of these shops sells only one model of shoe. We want to do some analysis of the average order value (AOV). When we look at orders data over a 30 day window, we naively calculate an AOV of $3145.13. Given that we know these shops are selling sneakers, a relatively affordable item, something seems wrong with our analysis.
#### Think about what could be going wrong with our calculation.
* Think about a better way to evaluate this data.
* What metric would you report for this dataset?
* What is its value?

In [39]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [3]:
path = ('/Users/fc-127/Desktop/data/2019 Winter Data Science Intern Challenge Data Set - Sheet1.csv')

In [5]:
df = pd.read_csv(path)

In [6]:
df.head()

order_id  shop_id  user_id  order_amount  total_items payment_method  \
0         1       53      746           224            2           cash   
1         2       92      925            90            1           cash   
2         3       44      861           144            1           cash   
3         4       18      935           156            1    credit_card   
4         5       18      883           156            1    credit_card   

            created_at  
0  2017-03-13 12:36:56  
1  2017-03-03 17:38:52  
2   2017-03-14 4:23:56  
3  2017-03-26 12:43:37  
4   2017-03-01 4:35:11

Quick EDA 

In [30]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5000 entries, 0 to 4999
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   order_id        5000 non-null   int64 
 1   shop_id         5000 non-null   int64 
 2   user_id         5000 non-null   int64 
 3   order_amount    5000 non-null   int64 
 4   total_items     5000 non-null   int64 
 5   payment_method  5000 non-null   object
 6   created_at      5000 non-null   object
dtypes: int64(5), object(2)
memory usage: 273.6+ KB


confirm no nulls and datatypes are appropriate

In [9]:
df.shape

(5000, 7)

5000 rows, 7 columns

In [11]:
df.created_at.min()


'2017-03-01 0:08:09'

In [12]:
df.created_at.max()

'2017-03-30 9:55:00'

confirm: over a "30 day window" 
identify: the 30 day period of the data: March 1- March 30 2017

In [17]:
df.shop_id.nunique()

100

confirm: "we have exactly 100 sneaker shops"

In [16]:
df.order_amount.describe()

count      5000.000000
mean       3145.128000
std       41282.539349
min          90.000000
25%         163.000000
50%         284.000000
75%         390.000000
max      704000.000000
Name: order_amount, dtype: float64

From .describe on order_amount column:
* we can identify the source of AOV (average order value): 3145.13
* mean across the entire dataset for value of order amounts
* can see outliers in max: 704000 
* std too high: meaning distribution of values very far from the mean
* 75 percent of of the values are under 390: 
* which again confirms that AOV of 3145 is off 
* we can deduct that data is NOT normally distributed 
* suggestion: choose a more approriate metric in the short term 
* median which is an alternate measure of central tendency
* more robust and less affected by the outliers in our dataset


In [36]:
df.total_items.value_counts()

2       1832
1       1830
3        941
4        293
5         77
2000      17
6          9
8          1
Name: total_items, dtype: int64

confirm source of outliers
* while the mjority of transactions involve transactiosn with 1 and 2 items 
* 17 rows have transactions of 2000 items which contributed to outliers * reseulting in max transaction amount $704,000 - not a good AOV for sneaker shop

In [40]:
df.order_amount.median()


284.0

## What metric would you report for this dataset?
* median
## What is its value? 
*  $284